service/src/main/java/de/adorsys/psd2/sandbox/xs2a/service/redirect/RedirectService.java

```diff
@@ -75,14 +75,18 @@ public void handleConsentCreationRedirectRequest(String externalId, String psuId
     AisConsent consent = aisConsentRepository.findOne(aisConsentAuth.getId());
 
     Optional<TestPsu> psu = testDataService.getPsu(psuId);
-    if (psu.isPresent()) {
-      Optional<ConsentStatus> consentStatus = ConsentStatus.fromValue(psu.get()
-          .getConsentStatusAfterSca().xs2aValue());
-      consentStatus.ifPresent(consent::setConsentStatus);
-      ScaStatus newScaStatus = ScaStatus.fromValue(psu.get().getInitiationScaStatus().xs2aValue());
-      aisConsentAuth.setScaStatus(newScaStatus);
 
+    Optional<ConsentStatus> consentStatus;
+    ScaStatus newScaStatus;
+    if (psu.isPresent()) {
+      consentStatus = ConsentStatus.fromValue(psu.get().getConsentStatusAfterSca().xs2aValue());
+      newScaStatus = ScaStatus.fromValue(psu.get().getInitiationScaStatus().xs2aValue());
+    } else {
+      consentStatus = ConsentStatus.fromValue("rejected");
+      newScaStatus = ScaStatus.fromValue("failed");
```

Comment:
You can set the TransactionStatus and ScaStatus easier without using fromValue. Just use the enum Values as described in the comments below.